File: python/plugin_config.py
class PluginConfig:
    """插件配置类"""
    
    def __init__(self):
        self.server_host = None
        self.server_port = None
        self.plugin_id = None
        self.plugin_name = None
        self.plugin_version = None
        self.plugin_type = None
        self.additional_config = {}
        
    def get_server_host(self):
        return self.server_host
    
    def set_server_host(self, server_host):
        self.server_host = server_host
        
    def get_server_port(self):
        return self.server_port
    
    def set_server_port(self, server_port):
        self.server_port = server_port
        
    def get_plugin_id(self):
        return self.plugin_id
    
    def set_plugin_id(self, plugin_id):
        self.plugin_id = plugin_id
        
    def get_plugin_name(self):
        return self.plugin_name
    
    def set_plugin_name(self, plugin_name):
        self.plugin_name = plugin_name
        
    def get_plugin_version(self):
        return self.plugin_version
    
    def set_plugin_version(self, plugin_version):
        self.plugin_version = plugin_version
        
    def get_plugin_type(self):
        return self.plugin_type
    
    def set_plugin_type(self, plugin_type):
        self.plugin_type = plugin_type
        
    def get_additional_config(self):
        return self.additional_config
    
    def set_additional_config(self, additional_config):
        self.additional_config = additional_config
        
    def add_config(self, key, value):
        self.additional_config[key] = value
        
    def get_config(self, key):
        return self.additional_config.get(key)

File: rust/src/plugin_config.rs
use std::collections::HashMap;

/// 插件配置结构体
#[derive(Debug, Clone)]
pub struct PluginConfig {
    server_host: String,
    server_port: i32,
    plugin_id: String,
    plugin_name: String,
    plugin_version: String,
    plugin_type: String,
    additional_config: HashMap<String, String>,
}

impl PluginConfig {
    pub fn new() -> Self {
        Self {
            server_host: String::new(),
            server_port: 0,
            plugin_id: String::new(),
            plugin_name: String::new(),
            plugin_version: String::new(),
            plugin_type: String::new(),
            additional_config: HashMap::new(),
        }
    }

    pub fn get_server_host(&self) -> &str {
        &self.server_host
    }

    pub fn set_server_host(&mut self, server_host: String) {
        self.server_host = server_host;
    }

    pub fn get_server_port(&self) -> i32 {
        self.server_port
    }

    pub fn set_server_port(&mut self, server_port: i32) {
        self.server_port = server_port;
    }

    pub fn get_plugin_id(&self) -> &str {
        &self.plugin_id
    }

    pub fn set_plugin_id(&mut self, plugin_id: String) {
        self.plugin_id = plugin_id;
    }

    pub fn get_plugin_name(&self) -> &str {
        &self.plugin_name
    }

    pub fn set_plugin_name(&mut self, plugin_name: String) {
        self.plugin_name = plugin_name;
    }

    pub fn get_plugin_version(&self) -> &str {
        &self.plugin_version
    }

    pub fn set_plugin_version(&mut self, plugin_version: String) {
        self.plugin_version = plugin_version;
    }

    pub fn get_plugin_type(&self) -> &str {
        &self.plugin_type
    }

    pub fn set_plugin_type(&mut self, plugin_type: String) {
        self.plugin_type = plugin_type;
    }

    pub fn get_additional_config(&self) -> &HashMap<String, String> {
        &self.additional_config
    }

    pub fn set_additional_config(&mut self, additional_config: HashMap<String, String>) {
        self.additional_config = additional_config;
    }

    pub fn add_config(&mut self, key: String, value: String) {
        self.additional_config.insert(key, value);
    }

    pub fn get_config(&self, key: &str) -> Option<&String> {
        self.additional_config.get(key)
    }
}